src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -122,8 +122,14 @@ void AnnotationManager::updateStyle(Style& style) {
         layer->sourceLayer = PointLayerID;
         layer->layout.icon.image = std::string("{sprite}");
         layer->layout.icon.allowOverlap = true;
+        layer->layerSpriteAtlas = atlas;
 
         style.addLayer(std::move(layer));
+    } else {
```

Comment:
Is the else branch necessary here? The only place that the point annotation layer gets added to the style is in the `if` branch above, so it shouldn't be possible for the style not to have the annotation source, but have a point annotation layer without its sprite atlas set.